test/e2e-api/kafka.go

```diff
@@ -30,46 +30,73 @@ import (
 
 const bootstrapServers = "kafka-broker:9092"
 
+var m sync.Mutex
+
 type KafkaController struct {
 	brokers     []string
 	adminClient sarama.ClusterAdmin
 	producer    sarama.SyncProducer
 	consumer    sarama.Consumer
 }
 
-func NewKafkaController() *KafkaController {
-	// initialize Kafka handlers
-	var brokers = []string{bootstrapServers}
+func initNewKafka(n *KafkaController) *KafkaController {
```

Comment:
Why it returns `*KafkaController`?

Reply:
i felt it was just cleaner to do that. that way when i call the function i can just re-assign the controller like this `kh = initNewKafka(kh)`. otherwise i would need to return producer, consumer and adminClient individually and re-assign those fields when the controller is called right ?